blas/gonum/level1double.go

```diff
@@ -591,7 +591,5 @@ func (Implementation) Dscal(n int, alpha float64, x []float64, incX int) {
 		f64.ScalUnitary(alpha, x[:n])
 		return
 	}
-	for ix := 0; ix < n*incX; ix += incX {
-		x[ix] *= alpha
-	}
+	f64.ScalInc(alpha, x, uintptr(n), uintptr(incX))
```

Comment:
This change requires a change to "single_precision.bash" and an invocation of `go generate`.

```diff
diff --git a/blas/gonum/single_precision.bash b/blas/gonum/single_precision.bash
index 0f5910d..4849482 100755
--- a/blas/gonum/single_precision.bash
+++ b/blas/gonum/single_precision.bash
@@ -23,6 +23,7 @@ cat level1double.go \
 | gofmt -r 'f64.AxpyUnitary -> f32.AxpyUnitary' \
 | gofmt -r 'f64.AxpyUnitaryTo -> f32.AxpyUnitaryTo' \
 | gofmt -r 'f64.DotUnitary -> f32.DotUnitary' \
+| gofmt -r 'f64.ScalInc -> f32.ScalInc' \
 | gofmt -r 'f64.ScalUnitary -> f32.ScalUnitary' \
 \
 | sed -e "s_^\(func (Implementation) \)D\(.*\)\$_$WARNING\1S\2_" \
```